Framework/Kernel/src/Unit.cpp

```diff
@@ -644,56 +644,80 @@ void dSpacing::init() {
       }
     }
   }
+  // force the assumption that difc is positive in debug builds
+  assert(difc > 0.);
 }
 
 double dSpacing::singleToTOF(const double x) const {
   if (!isInitialized())
     throw std::runtime_error("dSpacingBase::singleToTOF called before object "
                              "has been initialized.");
-  return difa * x * x + difc * x + tzero;
+  if (difa == 0.)
```

Comment:
does c++ support float point equal comparison?

Reply:
Yes it does. Comparing to most (non-integer) float values normally gets you into the realm of precision. However, this is checking against the value that the constructor sets as the default.